File: backend/src/main/java/com/whiteblue/backend/domain/application/entity/Application.java
package com.whiteblue.backend.domain.application.entity;

import com.whiteblue.backend.domain.user.entity.User;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@EntityListeners(AuditingEntityListener.class)
@Entity
public class Application {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String name;

    private String phoneNumber;

    private String major;

    private String introduction;

    @CreatedDate
    private LocalDateTime createdDate;

    @LastModifiedDate
    private LocalDateTime modifiedDate;

    @NotBlank
    @OneToOne(cascade = CascadeType.PERSIST, fetch = FetchType.LAZY)
    private User writer;

    @Builder
    public Application(Integer id, String name, String phoneNumber, String major, String introduction, User writer) {
        this.id = id;
        this.name = name;
        this.phoneNumber = phoneNumber;
        this.major = major;
        this.introduction = introduction;
        this.writer = writer;
    }
}

File: backend/src/main/kotlin/com/whiteblue/backend/domain/application/model/Application.kt
package com.whiteblue.backend.domain.application.model

import com.whiteblue.backend.domain.user.model.User
import com.whiteblue.backend.global.common.BaseEntity
import jakarta.persistence.*

@Table(name = "application")
@Entity
class Application(
    var name: String,

    var phoneNumber: String,

    var major: String,

    var introduction: String,

    @OneToOne(cascade = [CascadeType.PERSIST], fetch = FetchType.LAZY)
    val writer: User
) : BaseEntity() {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    val id: Long? = null
}
